src/main/java/pl/psnc/dei/queue/TasksQueue.java

```diff
@@ -97,7 +97,7 @@ private void processingFailed() throws InterruptedException {
 		}
 	}
 
-	public void addToQueue(Task task) {
+	public synchronized void addToQueue(Task task) {
```

Comment:
there could be a deadlock with line 60

Reply:
In my opinion, there won't be deadlock here (tested and consulted :)). Anyway, Krystian will change the implementation of the queue as part of merging all changes into one that does not require creating a synchronized block (BlockingQueue)